ui/pages/onboarding-flow/welcome/welcome.test.js

```diff
@@ -122,5 +126,52 @@ describe('Onboarding Welcome Component', () => {
         expect(mockHistoryPush).toHaveBeenCalledWith(ONBOARDING_METAMETRICS);
       });
     });
+
+    describe('useEffect for adding networks', () => {
+      it('should add networks when onboarding is incomplete', async () => {
+        // Render the component
+        renderWithProvider(<OnboardingWelcome />, mockStore);
+
+        await waitFor(() => {
+          // Verify addNetwork is called with Arbitrum (or other networks)
+          expect(addNetwork).toHaveBeenCalledWith({
+            chainId: '0xa4b1', // Arbitrum's chain ID
+            blockExplorerUrls: ['https://explorer.arbitrum.io'], // Arbitrum block explorer
+            defaultRpcEndpointIndex: 0,
+            defaultBlockExplorerUrlIndex: 0,
+            name: 'Arbitrum One',
+            nativeCurrency: 'ETH',
+            rpcEndpoints: [
+              {
+                url: 'https://arbitrum-mainnet.infura.io/v3/undefined',
+                type: 'custom',
+              },
+            ],
+          });
+        });
+      });
+
+      it('should not add networks when onboarding is completed', async () => {
+        const mockCompletedOnboardingState = {
+          ...mockState,
+          metamask: {
+            ...mockState.metamask,
+            completedOnboarding: true, // Simulate completed onboarding
+          },
+        };
+        const mockStore2 = configureMockStore([thunk])(
+          mockCompletedOnboardingState,
+        );
+
+        renderWithProvider(<OnboardingWelcome />, mockStore2);
+
+        // Wait to ensure the effect runs (or doesn't in this case)
+        await waitFor(() => {
+          // Expect that neither addNetwork nor updateNetworksList was called
+          expect(addNetwork).not.toHaveBeenCalled();
+          expect(updateNetworksList).not.toHaveBeenCalled();
+        });
+      });
+    });
```

Comment:
Thanks for adding tests 🙏 
Can we add a unit test to cover the case where await dispatch(addNetwork) fails?

Reply:
fixed [36d9b48](https://github.com/MetaMask/metamask-extension/pull/28006/commits/36d9b4834d65dd0ff27b45d376d10feb8afbd683)